packages/@aws-cdk/aws-location-alpha/lib/route-calculator.ts

```diff
@@ -0,0 +1,158 @@
+import * as iam from 'aws-cdk-lib/aws-iam';
+import { ArnFormat, IResource, Lazy, Resource, Stack, Token } from 'aws-cdk-lib/core';
+import { Construct } from 'constructs';
+import { CfnRouteCalculator } from 'aws-cdk-lib/aws-location';
+import { DataSource } from './place-index';
+import { generateUniqueId } from './util';
+
+/**
+ * A Route Calculator
+ */
+export interface IRouteCalculator extends IResource {
+  /**
+   * The name of the route calculator
+   *
+   * @attribute
+   */
+  readonly routeCalculatorName: string;
+
+  /**
+   * The Amazon Resource Name (ARN) of the route calculator resource
+   *
+   * @attribute Arn,CalculatorArn
+   */
+  readonly routeCalculatorArn: string;
+}
+
+/**
+ * Properties for a route calculator
+ */
+export interface RouteCalculatorProps {
+  /**
+   * A name for the route calculator
+   *
+   * @default - A name is automatically generated
+   */
+  readonly routeCalculatorName?: string;
+
+  /**
+   * Data source for the route calculator
+   *
+   * @default DataSource.ESRI
+   */
+  readonly dataSource?: DataSource;
```

Comment:
Currently, the DataSource for `PlaceIndex` and RouteCalculator are identical.
Therefore, I decided to reference the DataSource of PlaceIndex.
However, if there's a better approach, please share your opinion.

Ref:
* https://docs.aws.amazon.com/location/latest/developerguide/what-is-data-provider.html
* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-location-placeindex.html#cfn-location-placeindex-datasource
* https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-location-routecalculator.html#cfn-location-routecalculator-datasource

I also considered defining a completely separate Enum like `RouteDataSource`, but decided against it as I thought it would be redundant.
I also thought about moving PlaceIndex's `DataSource` to a separate file, but I think it would result in breaking changes...

Reply:
Agree. Let's see if the core team has a different opinion.

Reply:
I see in another PR of yours that you added a `util` file. I think it would be nice to move PlaceIndex's `DataSource` to somewhere more generic. Based on the CFN documentation, https://docs.aws.amazon.com/location/latest/developerguide/what-is-data-provider.html, it seems that all these three resources should re-use the `DataSource` prop.

> but I think it would result in breaking changes
I don't see why this would be a breaking change. Can you explain that to me please?

Reply:
Sorry, I misunderstood the breaking change. Although I needed to fix [unit test within the package](https://github.com/aws/aws-cdk/pull/30682/files#diff-19b3113c1cea38d5a1a0ac770bebce60582e51297b1bdcdcc397dbdf7d7443b6R5),  it doesn't actually affect the usage of the package.

I've moved the DataSource to the util module.